packages/davinci-client/src/lib/collector.types.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * @interface SingleValueCollector - Represents a request to collect a single value from the user, like email or password.
+ */
+export type SingleValueCollectorTypes =
+  | 'TextCollector'
+  | 'PasswordCollector'
+  | 'ListCollector'
+  | 'SingleValueCollector';
+export interface SingleValueCollector {
```

Comment:
I think making this generic may help improve inference.

```ts
export interface SingleValueCollector<T extends SingleValueCollectorTypes = 'SingleValueCollector'> { 
...
type: T
...
}
```

We can have a default generic value as above in case we dont want to trickle down all the code where this is used.

https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBDAnmYcDKBLAdgcwDbABqAhngK7ADCEeBAxjNACrLADOcAvAFBxwA+cAORNQMarWANoQ3gOEAFYmzYB3aABMJ9RlFl9BQgDIY24mjplzDmXARLkqFqbqEBubqEiw42GMCgAM2I6VFt8IlIKbRdoAB4mODFgLA0OcPsop0lpKBYUNgA+OABvOT44OmJ-HGhEAC5hDMjHGNz3OSQURsTk1I5sQIC4BTgAfhG4RqFmh2jndrkMDUazKGwcDz4sYgBbYFWYddwt3ywwMhhGsoq4AGtgBrg1jdO+ADcsw+OceSwyXYAI2GgkBEBowGIWDeCFY31ecgAvqcIJcLldSuV7o94ScsXhiMC8LjNliugdnkcEbdPo4xiTTsjuIjuNw6BAsGY4BBAQArRqzLJtXRxERiYUyQqsnm8gB05O4AHpFRUAHpjIA

added here for more visual